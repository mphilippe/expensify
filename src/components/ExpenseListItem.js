import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';
import { removeExpense } from '../actions/expenses';
import '../config/numeral.js';

export const ExpenseListItem = ({ id, description, note, amount, createdAt, dispatch }) => (
  <div>
    <h3>
      <Link to={`/edit/${id}`}>{description}</Link>
    </h3>
    {note && <p>{note}</p>}
    <p>
      {numeral(amount / 100).format('$0,0[.]00')}
      -
      {moment(createdAt).format('DD-MM-YYYY')}
    </p>
    <button onClick={(e) => {
      dispatch(removeExpense({
        id
      }));
    }}>Remove</button>
  </div>
);

export default connect()(ExpenseListItem);
