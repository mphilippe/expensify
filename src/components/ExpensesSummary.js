import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import '../config/numeral.js';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';

// export const ExpensesSummary = (props) => (
//   <div>
//     {props.expenses.length === 0 ? (
//       <p>No expense</p>
//     ) : (
//       <p>
//         Viewing {props.expenses.length} 
//         expense{props.expenses.length > 1 ? 's' : ''} 
//         totalling 
//         {numeral(expensesTotal(props.expenses) / 100).format('$0,0[.]00')}
//       </p>
//     )}
//   </div>
// );

// const mapStateToProps = (state) => ({
//   expenses: selectExpenses(state.expenses, state.filters)
// });

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = 'expense' + (expenseCount > 1 ? 's' : '');
  const formattedTotal = numeral(expensesTotal / 100).format('$0,0[.]00');
  return (
    <div>
      {expenseCount === 0 ? (
        <p></p>
      ) : (
        <p>
          Viewing {expenseCount} {expenseWord} totalling {formattedTotal}
        </p>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: expensesTotal(visibleExpenses)
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
