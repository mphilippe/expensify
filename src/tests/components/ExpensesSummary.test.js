import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';

test('Should render ExpensesSummary without expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={0}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with 1 expense correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={60000}/>);
  expect(wrapper).toMatchSnapshot();
});

test('Should render ExpensesSummary with many expenses correctly', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={2160050}/>);
  expect(wrapper).toMatchSnapshot();
});
