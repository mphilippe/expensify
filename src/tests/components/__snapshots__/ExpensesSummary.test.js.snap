// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary with 1 expense correctly 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    ₱600
  </p>
</div>
`;

exports[`Should render ExpensesSummary with many expenses correctly 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling 
    ₱21,600.50
  </p>
</div>
`;

exports[`Should render ExpensesSummary without expense correctly 1`] = `
<div>
  <p />
</div>
`;
